sqlglot/parser.py

```diff
@@ -522,6 +523,7 @@ class Parser(metaclass=_Parser):
         *TYPE_TOKENS,
         *NO_PAREN_FUNCTIONS,
     }
+    ID_VAR_TOKENS.remove(TokenType.UNION)
```

Comment:
Should this be moved to DuckDB's parser? Although it's true for most of the dialects that I tested, I see that (at least) Spark does not have an issue:

```
spark-sql (default)> with t as (select 1 union) select union from t;
union
1
```

Reply:
This was intentional, we don't support `union` as an identifier today in main so this PR is not breaking any existing functionality. I think it's out of scope for now to support `UNION` as an identifier token.